docs: :building_construction: clarify multi-deposit repos and the input metadata files (#56)
seedcase-project/zen-do: docs/design/interface/inputs.qmd

This file contains the metadata for the deposit. It should represent the
"source of truth", as any action taken by zen-do when interacting with
the Zenodo API will use this file and overwrite anything in the Zenodo
deposit.

::: callout-note
For the `convert` command, the input is actually a format other than the
`.zenodo.toml` file, such as the `.zenodo.json` file. But the same or
similar metadata is contained in those other formats.
:::

> For repositories that have or will have multiple deposits, then there

Reviewer: ```suggestion
For repositories that have or will have multiple deposits, there
```

will also be multiple `.zenodo.toml` files, one for each deposit. For
most use-cases, you would store the `.zenodo.toml` file in the folder

> that links it to the file that will be uploaded to the deposit. For

Reviewer: ```suggestion
most use-cases, you would store the `.zenodo.toml` file in the same folder
as the file that will be uploaded to the deposit as its content. For
```

> example, for a research project repository, there may be a deposit for

Reviewer: ```suggestion
example, in a research project repository, there may be a deposit for
```

the paper with the metadata kept in `docs/paper/.zenodo.toml` alongside
the paper PDF in `docs/paper/paper.pdf` and a deposit for the data with
the metadata kept in `data/.zenodo.toml` for the data files in `data/`.
This way, the metadata is kept close to the content it describes.

> However, sometimes multiple deposits might describe content in the same

Reviewer: Okay! So we support both separate folders and separate names.
And based on the other PRs, when we take the metadata as input, we still default to `.zenodo.toml` in the current dir, but they can specify another one.

Which rule should `init` follow when creating a metadata file? Same defaults as when reading and error if file already exists?

Author: Not sure what you mean with "same defaults as when reading", but `init` would output an empty `.zenodo.toml` in the working directory, but won't overwrite anything. Does that make sense?

Author: If the user wants, they can rename after creating it.

Reviewer: I meant what if there is already a `.zenodo.toml` in the working directory? Throw an error?

folder. For example, in the case of a repository with teaching material
that builds into a website (as HTML) and as a PDF, the metadata file for
the repository source files would be `.zenodo.toml` in the root of the
repository, while the metadata for the PDF (that's rendered from the
root source files) could be either stored in another folder or be called
`.pdf.zenodo.toml` (or `.book.zenodo.toml` if the output is a book) in
the root of the repository. In this case, the metadata files are kept in

> the same location but with different names to reflect their contents.

Reviewer: ```suggestion
the same location but have different names to reflect their contents.
```

## Zenodo access token

This is required in order to interact with the Zenodo API. Users can't
directly give zen-do this token. For security purposes, zen-do doesn't
accept the token as an argument since that would be visible in the
terminal history and process list. Instead, zen-do will look for a token
with the name `ZENODO_TOKEN` (or `ZENODO_SANDBOX_TOKEN`) in the system
keyring first and, if not found, in the environment variables.

We only allow the use of an environment variable because continuous
deployment platforms like GitHub Actions have built-in support for
secret environment variables, and this is a common way to store and use
secrets in these types of pipelines. Outside of these deployment
platforms, we recommend users to store their token in their system
keyring, which is a more secure way to store secrets.

## Files

These are the files that will be attached to a Zenodo deposit that the
metadata is describing. These files will be be uploaded along with the
metadata in `.zenodo.toml` to the Zenodo deposit. Any file that is

> uploadable to Zenodo can be accepted.

Reviewer: ```suggestion
uploadable to Zenodo is accepted.
```
